src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -258,6 +298,39 @@ def get(self, request: Request, project: Project) -> Response:
                     # When no code mapping have been matched we have not attempted a URL
                     if current_config["outcome"].get("attemptedUrl"):
                         result["attemptedUrl"] = current_config["outcome"]["attemptedUrl"]
+
+                # Get commit sha from Git blame
+                try:
+                    should_get_commit_sha = (
+                        (
+                            not ctx["commit_id"]
+                            or ctx["commit_id"] == current_config["config"]["defaultBranch"]
+                        )
+                        and features.has(
+                            "organizations:codecov-stacktrace-integration",
+                            project.organization,
+                            actor=request.user,
+                        )
+                        and project.organization.flags.codecov_access
+                    )
```

Comment:
nit: this might be more readable if you broke this up. Also helps with short circuit evaluation.

```suggestion
                    codecov_enabled = features.has(
                            "organizations:codecov-stacktrace-integration",
                            project.organization,
                            actor=request.user,
                        )
                        and project.organization.flags.codecov_access
                    should_get_commit_sha = codecov_enabled and (
                            not ctx["commit_id"]
                            or ctx["commit_id"] == current_config["config"]["defaultBranch"]
                        )
```